src/osha/oira/client/browser/client.py

```diff
@@ -230,13 +266,32 @@ def get_token(self):
         return token
 
     def get_addresses_for_groups(self, group_paths):
-        subscribers = (
+        subscribers = []
+        other = []
+        for group_id in group_paths:
+            if "/" in group_id or "-" not in group_id:
+                other.append(group_id)
+                continue
+            # Special handling for language specific mailing lists, e.g. "be-fr"
+            country, lang = group_id.split("-")
+            country_subscribers = (
+                Session.query(Account.loginname)
+                .filter(Account.id == NewsletterSubscription.account_id)
+                .filter(NewsletterSubscription.zodb_path == (country))
+                .filter(Account.id == NewsletterSetting.account_id)
+                .filter(NewsletterSetting.value == f"language:{lang}")
+                .group_by(Account.loginname)
+            )
+            subscribers.extend([s.loginname for s in country_subscribers])
+
+        other_subscribers = (
             Session.query(Account.loginname)
             .filter(Account.id == NewsletterSubscription.account_id)
-            .filter(NewsletterSubscription.zodb_path.in_(group_paths))
+            .filter(NewsletterSubscription.zodb_path.in_(other))
             .group_by(Account.loginname)
         )
-        return [s.loginname for s in subscribers]
+        subscribers.extend([s.loginname for s in other_subscribers])
+        return subscribers
 
     @property
     def results(self):
```

Comment:
Should this be memoized? Maybe out of scope. But I've seen a `memoize` on a `getPhysicalPath`. The memoize logic is also quite a chunk of code where I'm **not sure** what is more expensive - memoizing everything vs memoizing selectively.

Whatever. As soon as this results method depends on request data (user session, etc) memoizing does not make sense at all.

Never mind.